scripts/populate_file_web_server.py

```diff
@@ -0,0 +1,308 @@
+                   PRINT_LEVEL_WARN     : 2,
+                   PRINT_LEVEL_INFO     : 3,
+                   PRINT_LEVEL_VERBOSE  : 4 }
+
+#return code
+RET_CODE_SUCCESS                = 0
+RET_CODE_CANNOT_CREATE_FILE     = -1
+RET_CODE_CANNOT_OPEN_FILE       = -2
+RET_CODE_HTTP_SERVER_ERROR      = -3
+RET_CODE_CANNOT_WRITE_FILE      = -4
+
+#constants
+RESOURCES_FILE_NAME       = 'versions-web'
+EXCLUDE_DIRECTORES        = ['fsroot', 'target']
+HASH_SEPARATOR            = '-'
+DEFAULT_INVALID_INPUT     = 'none'
+
+# global variables
+g_current_print_level     = PRINT_LEVEL_INFO
+
+#Script debug features (disabled by default)
+g_delete_resources_in_cache = True
+
+
+# global Classes
+class Resource:
+    def __init__(self, line, file):
+        self.file = file
+        temp=line.split("==")
+        assert(2==len(temp))
+        self.url=temp[0].strip()
+        self.hash=temp[1].strip()
+        temp=self.url.split("/")
+        assert(len(temp)>0)
+        self.name=temp[len(temp)-1]
+        #handle special scenarios
+        if 0 != self.name.count('?') == True:
+            temp = self.name.split("?")
+            self.name = temp[0]
+
+    def get_unique_name(self):
+        return self.name + HASH_SEPARATOR + self.hash
+
+    def get_url(self):
+        return self.url
+
+    def __str__(self):
+        ret_val = "Resource name: " + self.name + "\n"
+        ret_val += "File: " + self.file + "\n"
+        ret_val += "Hash: " + self.hash + "\n"
+        ret_val += "Full URL: " + self.url
+        return ret_val
+
+
+
+# Helper functions
+
+def print_msg(print_level, msg, print_in_place=False):
+    if PRINT_LEVEL_LUT[g_current_print_level] >= PRINT_LEVEL_LUT[print_level]:
+        if True == print_in_place:
+            print(msg, end='\r')
+        else:
+            print(msg)
+
+def create_dir_if_not_exist(dir):
+    if not os.path.exists(dir):
+        try:
+            os.mkdir(dir)
+        except:
+            print_msg(PRINT_LEVEL_WARN, "Cannot create directory " + dir)
+
+def delete_file_if_exist(file):
+
+    if os.path.exists(file):
+        try:
+            os.remove(file)
+        except:
+            print_msg(PRINT_LEVEL_WARN, "Cannot delete " + file)
+        
+# Logic functions
+        
+def generate_output_file(resources, dest_url_valid, dest_url, output_file_name):
+
+    try:
+        with open(output_file_name, 'w') as f:
+            for unique_name in resources.keys():
+                resource = resources[unique_name]
+                if True == dest_url_valid:
+                    line = dest_url
+                else:
+                    line = resource.get_url()
+                if line[-1] != '/':
+                    line += '/'
+                line += resource.name + "==" + resource.hash
+                f.write(line + '\n')
+    except:
+        print_msg(PRINT_LEVEL_WARN, output_file_name + " cannot be created")
+        return RET_CODE_CANNOT_CREATE_FILE
+        
+    f.close()
```

Comment:
f.close() can be removed, because the "with open" used.

Reply:
Good idea. New version was uploaded.